docs: add transitioning docs for `poetry` and `conda` (#1624)
prefix-dev/pixi: docs/switching_from/conda.md

    ```shell
    ~/myenv > eval "$(pixi shell-hook)"
    ```

## Environment vs Project
`Conda` and `mamba` focus on managing environments, while `pixi` emphasizes projects.
In `pixi`, a project is a folder containing a [manifest](../reference/project_configuration.md)(`pixi.toml`/`pyproject.toml`) file that describes the project, a `pixi.lock` lock-file that describes the exact dependencies, and a `.pixi` folder that contains the environment.

This project-centric approach allows for easy sharing and collaboration, as the project folder contains all the necessary information to recreate the environment.
It manages more than one environment for more than one platform in a single project, and allows for easy switching between them. (See [multiple environments](../features/multi_environment.md))

## Global environments
`conda` installs all environments in one global location.
When this is important to you for filesystem reasons, you can use the [detached-environments](../reference/pixi_configuration.md#detached-environments) feature of pixi.
```shell
pixi config set detached-environments true
# or a specific location
pixi config set detached-environments /path/to/envs
```

> This doesn't allow you to activate the environments using `pixi shell -n` but it will make the installation of the environments go to the same folder.

Reviewer: I might move this to troubleshooting, as having it here makes it sound more like a suggestion on the best way for a Conda user to use Pixi.

`pixi` does have the `pixi global` command to install tools on your machine. (See [global](../reference/cli.md#global))
This is not a replacement for `conda` but works the same as [`pipx`](https://pipx.pypa.io/stable/) and [`condax`](https://mariusvniekerk.github.io/condax/).
It creates a single isolated environment for the given requirement and installs the binaries into the global path.
```shell
pixi global install bat
bat pixi.toml
```

!!! warn "Never install `pip` with `pixi global`"
Installations with `pixi global` get their own isolated environment.
Installing `pip` with `pixi global` will create a new isolated environment with its own `pip` binary.
Using that `pip` binary will install packages in the `pip` environment, making it unreachable form anywhere as you can't activate it.

> ## Automated switching

Reviewer: `Automated switching` feels like it describes Pixi using the right task environment over moving from `environment.yml`.

```suggestion
## Migrating from environment.yml
```

# Transitioning from the `conda` or `mamba` to `pixi`
Welcome to the guide designed to ease your transition from `conda` or `mamba` to `pixi`.
This document compares key commands and concepts between these tools, highlighting `pixi`'s unique approach to managing environments and packages.
With `pixi`, you'll experience a project-based workflow, enhancing your development process, and allowing for easy sharing of your work.

## Why Pixi?

> `Pixi` builds upon the foundation of the conda ecosystem, introducing a project-centric approach rather than focusing solely on environments.

Reviewer: ```suggestion
`Pixi` builds upon the foundation of the conda ecosystem, introducing a project-centric approach rather than focusing solely on named global environments.
```

> This shift towards projects offers a more organized and efficient way to manage dependencies and run code, tailored to modern development practices.

Reviewer: Maybe specify that the dependency management is automated rather than manual?
```suggestion
This shift towards projects offers a more automated, organized, and efficient way to manage dependencies and run code, tailored to modern development practices.
```

| Creating an Environment     | `conda create -n myenv -c conda-forge python=3.8` | `pixi init myenv` followed by `pixi add python=3.8`                  |
| Activating an Environment   | `conda activate myenv`                            | `pixi shell` within the project directory                            |
| Deactivating an Environment | `conda deactivate`                                | `exit` from the `pixi shell`                                         |
| Running a Task              | `conda run -n myenv python my_program.py`         | `pixi run python my_program.py` (See [run](../reference/cli.md#run)) |
| Installing a Package        | `conda install numpy`                             | `pixi add numpy`                                                     |
| Uninstalling a Package      | `conda remove numpy`                              | `pixi remove numpy`                                                  |

!!! warn "No `base` environment"
    Conda has a base environment, which is the default environment when you start a new shell.
    **Pixi does not have a base environment**. And requires you to install the tools you need in the project or globally.

>     Using `pixi global install bat` will install `bat` in a global environment, which is not the same as the `base` environment in conda.

Reviewer: ```suggestion
    Using `pixi global install bat` will install `bat` in an isolated global environment (similar to pipx), which is not the same as the `base` environment in conda.
```

??? tip "Activating pixi environment in the current shell"
    For some advanced use-cases, you can activate the environment in the current shell.
    This uses the `pixi shell-hook` which prints the activation script, which can be used to activate the environment in the current shell without `pixi` itself.
    ```shell

>     ~/myenv > eval "$(pixi shell-hook)"

Reviewer: I'd emphasize `pixi shell`, possibly with it's own section over using `shell-hook`. 

It would also be worth showing how `--manifest-path` can be used to point to activate a shell from a different directory.

    ```

## Environment vs Project
`Conda` and `mamba` focus on managing environments, while `pixi` emphasizes projects.
In `pixi`, a project is a folder containing a [manifest](../reference/project_configuration.md)(`pixi.toml`/`pyproject.toml`) file that describes the project, a `pixi.lock` lock-file that describes the exact dependencies, and a `.pixi` folder that contains the environment.

This project-centric approach allows for easy sharing and collaboration, as the project folder contains all the necessary information to recreate the environment.

> It manages more than one environment for more than one platform in a single project, and allows for easy switching between them. (See [multiple environments](../features/multi_environment.md))

Reviewer: Maybe a callout on how Pixi can lock for multiple platforms to help enable collaboration to highlight Pixi's additional abilities?
